src/Sulu/Bundle/MediaBundle/Resources/js/containers/MediaSelection/tests/MediaSelection.test.js

```diff
@@ -0,0 +1,384 @@
+                    description: 'Description 1',
+                },
+                {
+                    id: 2,
+                    title: 'Title 2',
+                    objectCount: 0,
+                    description: 'Description 2',
+                },
+            ];
+
+            const thumbnails = {
+                'sulu-260x': 'http://lorempixel.com/260/100',
+                'sulu-25x25': 'http://lorempixel.com/25/25',
+            };
+
+            const mediaData = [
+                {
+                    id: 1,
+                    title: 'Title 1',
+                    mimeType: 'image/png',
+                    size: 12345,
+                    url: 'http://lorempixel.com/500/500',
+                    thumbnails: thumbnails,
+                },
+                {
+                    id: 2,
+                    title: 'Title 2',
+                    mimeType: 'image/jpeg',
+                    size: 54321,
+                    url: 'http://lorempixel.com/500/500',
+                    thumbnails: thumbnails,
+                },
+            ];
+
+            this.loading = false;
+            this.pageCount = 3;
+            this.data = (resourceKey === COLLECTIONS_RESOURCE_KEY)
+                ? collectionData
+                : mediaData;
+            this.selections = [];
+            this.getPage = jest.fn().mockReturnValue(2);
+            this.getFields = jest.fn().mockReturnValue({
+                title: {},
+                description: {},
+            });
+            this.destroy = jest.fn();
+            this.sendRequest = jest.fn();
+            this.clearSelection = jest.fn();
+            this.setAppendRequestData = jest.fn();
+            this.deselectEntirePage = jest.fn();
+            this.getSchema = jest.fn().mockReturnValue({});
+        }),
+    };
+});
+
+jest.mock('sulu-admin-bundle/containers/Datagrid/registries/DatagridAdapterRegistry', () => {
+    const getAllAdaptersMock = jest.fn();
+
+    return {
+        getAllAdaptersMock: getAllAdaptersMock,
+        add: jest.fn(),
+        get: jest.fn((key) => getAllAdaptersMock()[key]),
+        has: jest.fn(),
+    };
+});
+
+jest.mock('../../../stores/CollectionStore', () => jest.fn(function() {
+    this.destroy = jest.fn();
+}));
+
+jest.mock('sulu-admin-bundle/services', () => ({
+    translate: function(key) {
+        switch (key) {
+            case 'sulu_media.all_media':
+                return 'All Media';
+            case 'sulu_media.copy_url':
+                return 'Copy URL';
+            case 'sulu_media.download_masterfile':
+                return 'Downoad master file';
+            case 'sulu_admin.page':
+                return 'Page';
+            case 'sulu_admin.of':
+                return 'of';
+            case 'sulu_admin.object':
+                return 'Object';
+            case 'sulu_admin.objects':
+                return 'Objects';
+        }
+    },
+}));
+
+beforeEach(() => {
```

Comment:
I would not place the `beforeEach` in the middle of multiple `jest.mock`, but move it after them instead.